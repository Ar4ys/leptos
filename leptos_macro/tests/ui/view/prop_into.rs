use leptos::*;

#[component]
fn Component(#[prop(into)] prop: MaybeSignal<String>) -> impl IntoView {
    _ = prop;
}

fn main() {
    view! {
        <Component prop=move || String::new() />
    };

    let prop = move || String::new();

    view! {
        <Component prop />
    };
}
