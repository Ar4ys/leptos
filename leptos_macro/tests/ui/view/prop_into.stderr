error[E0277]: the trait bound `leptos::MaybeSignal<std::string::String>: From<{closure@$DIR/tests/ui/view/prop_into.rs:10:25: 10:32}>` is not satisfied
  --> tests/ui/view/prop_into.rs:10:25
   |
10 |         <Component prop=move || String::new() />
   |                    ---- ^^^^^^^^^^^^^^^^^^^^^ the trait `From<{closure@$DIR/tests/ui/view/prop_into.rs:10:25: 10:32}>` is not implemented for `leptos::MaybeSignal<std::string::String>`
   |                    |
   |                    required by a bound introduced by this call
   |
   = help: the trait `From<&str>` is implemented for `leptos::MaybeSignal<std::string::String>`
   = help: for that trait implementation, expected `&str`, found `{closure@$DIR/tests/ui/view/prop_into.rs:10:25: 10:32}`
   = note: required for `{closure@$DIR/tests/ui/view/prop_into.rs:10:25: 10:32}` to implement `Into<leptos::MaybeSignal<std::string::String>>`
note: required by a bound in `ComponentPropsBuilder::prop`
  --> tests/ui/view/prop_into.rs:3:1
   |
3  | #[component]
   | ^^^^^^^^^^^^ required by this bound in `ComponentPropsBuilder::prop`
4  | fn Component(#[prop(into)] prop: MaybeSignal<String>) -> impl IntoView {
   |                            ---- required by a bound in this associated function
   = note: this error originates in the derive macro `::leptos::typed_builder_macro::TypedBuilder` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `leptos::MaybeSignal<std::string::String>: From<{closure@$DIR/tests/ui/view/prop_into.rs:13:16: 13:23}>` is not satisfied
  --> tests/ui/view/prop_into.rs:16:20
   |
16 |         <Component prop />
   |                    ^^^^ the trait `From<{closure@$DIR/tests/ui/view/prop_into.rs:13:16: 13:23}>` is not implemented for `leptos::MaybeSignal<std::string::String>`
   |
   = help: the trait `From<&str>` is implemented for `leptos::MaybeSignal<std::string::String>`
   = help: for that trait implementation, expected `&str`, found `{closure@$DIR/tests/ui/view/prop_into.rs:13:16: 13:23}`
   = note: required for `{closure@$DIR/tests/ui/view/prop_into.rs:13:16: 13:23}` to implement `Into<leptos::MaybeSignal<std::string::String>>`
note: required by a bound in `ComponentPropsBuilder::prop`
  --> tests/ui/view/prop_into.rs:3:1
   |
3  | #[component]
   | ^^^^^^^^^^^^ required by this bound in `ComponentPropsBuilder::prop`
4  | fn Component(#[prop(into)] prop: MaybeSignal<String>) -> impl IntoView {
   |                            ---- required by a bound in this associated function
   = note: this error originates in the derive macro `::leptos::typed_builder_macro::TypedBuilder` (in Nightly builds, run with -Z macro-backtrace for more info)
